README.md

```diff
@@ -1,229 +1,894 @@
+most relevant data and avoid distracting and non-essential noise.
+
+Last, but not least, two more arguments: `-S` (`--success-stacks`) and `-T`
+(`--trace`).
+
+`-S` instructs `retsnoop` to not ignore "successful cases", i.e., those that
+don't end up returning errors. In our case we wanted to investigate
+successfully verified BPF program, so not specifying `-S` would completely
+ignore such case. Adding `-S` disables `retsnoop`'s default error-focused
+capturing logic.
+
+`-T` instructs `retsnoop` to collect a detailed record of all traced function
+calls and emit it in a separate table, like the one you can see above.
+
+We'll look at what that function call trace table shows in a second. But first,
+we do also see a standard stack trace at the very bottom of the output. In this
+case, it ends up with `bpf_map_kptr_off_contains` that returns `NULL`. This is
+`retsnoop` trying to be helpful to detect deepest error-returning function call
+chain (and `NULL` is considered an error by default). We don't care much about
+call stack in this example, so just ignore it in this case. Stack trace is
+always emitted by `retsnoop`, there is no way to suppress it. This might be
+possible in the future.
+
+Back to function call trace table. Let's shorten it some more to explain main
+elements:
 
-```shell
-$ sudo retsnoop -e '*_sys_write' -a ':fs/btrfs/*.c' -v
-[...]
+```
+FUNCTION CALLS TRACE                              RESULT                 DURATION
+-----------------------------------------------   --------------------  ---------
+→ bpf_prog_load
+    → bpf_prog_alloc
+        ↔ bpf_prog_alloc_no_stats                 [0xffffc9000031e000]    5.539us
+    ← bpf_prog_alloc                              [0xffffc9000031e000]   10.265us
+    [...]
+    → bpf_prog_kallsyms_add
+        ↔ bpf_ksym_add                            [void]                  2.046us
+    ← bpf_prog_kallsyms_add                       [void]                  6.104us
+← bpf_prog_load                                   [5]                   374.697us
 ```
 
-## Use cases
+In the left column we see function names with one of `→`, `←`, or `↔` markers
+on the left. `→` marks entry into traced function, `←` marks and exit from a
+function. If a function didn't have any further calls to any other traced
+function and thus is a leaf function (as far as a specified set of functions of
+interest goes; note, that in reality there could have been other functions
+called further, but they were not traced and thus captured by `retsnoop` due to
+them being omitted from `-e`/`-d` specifications). Such leaf functions, instead
+of taking two rows with separate `→` (entry) and `←` (exit) records, are
+collapsed into a single entry-exit row marked with `↔`. As calls become more
+nested, `retsnoop` indents each level of nestedness with four spaces.
+
+For each row corresponding to function exit event (so, `→`- and `↔`-marked
+rows), retsnoop also reports return result, similarly to default stack trace
```

Comment:
s/reports return result/reports the return result

also

s/similarly to default/similarly to the default